### ISSUE TITLE
Solving or documenting encoding issues in .NET Core

### ISSUE THREAD

**Issue by gabriele-tomassetti** [AngleSharp/AngleSharp#534]
**First thanks to the authors for this fantastic library.** One thing that I really like is that the API makes really easy to port code from Javascript libraries. Also the detection of encoding is very useful, but I found an issue on .NET Core.

.NET Core doesn't include support for some encodings by default. This lead to the problem that when AngleSharp tries to detect the encoding a lot of old content is forced to default to UTF-8.
The simple way to solve it, as [noted here](http://stackoverflow.com/questions/37870084/net-core-doesnt-know-about-windows-1252-how-to-fix) is:

> to include "System.Text.Encoding.CodePages" as a dependency
> and then call
> Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);

If it's not desirable to add this dependency and the dependencies of the [System.Text.Encoding.CodePages](https://www.nuget.org/packages/System.Text.Encoding.CodePages/) package itself, it may be useful to document the problem.

**Comment by FlorianRappl** [comment]
I think for .NET Core the dependency makes perfect sense.

Could you - by any chance - create a PR? As v0.10 is still in the pipe (shame on me!) the PR could make it in time.

**Comment by gabriele-tomassetti** [comment]
@FlorianRappl For the PR, I just need to clone master, create a branch for the issue and modify the "netstandard1.0" -> "dependencies" section in project.json, isn't it?

**Comment by jogibear9988** [comment]
you can also edit the file directly in github, it will fork, create a branch & pull req automaticaly

**Comment by FlorianRappl** [comment]
Please clone `devel`. `master` is the last release, `devel` is the current (in-development) version.

Also when submitting the PR please do it against `devel`! Thanks 👍 .

**Comment by LeMoussel** [comment]
@jogibear9988 How do you open/edit the file directly in github?

**Comment by jogibear9988** [comment]
https://help.github.com/articles/editing-files-in-your-repository/

**Comment by gabriele-tomassetti** [comment]
I have added the pull request #535 but I didn't notice that adding the System.Text.Encoding.CodePages package would also require to upgrade to netstandard1.3, from netstandard1.0.

**Comment by FlorianRappl** [comment]
Yep that is now a big question. I need to see what platforms would be excluded by such a change!

**Comment by gabriele-tomassetti** [comment]
![immagine](https://cloud.githubusercontent.com/assets/12112299/22897121/ca17e3c4-f222-11e6-9a90-06faacb0c6ef.png)
From [Introducing .NET Standard](https://blogs.msdn.microsoft.com/dotnet/2016/09/26/introducing-net-standard/).

If I am reading this correctly this would exclude Windows (Phone) 8 and the .NET Framework 4.5.*

Also for reference the [.NET Standard 1.0](https://github.com/dotnet/standard/blob/master/docs/versions/netstandard1.0.md) and [.NET Standard 1.3](https://github.com/dotnet/standard/blob/master/docs/versions/netstandard1.3.md)

**Comment by FlorianRappl** [comment]
Yep, I think so as well. But actually, that may not be much of a problem, as these two platforms are covered with the contained PCL build.

So we ship:
* PCL Build (.NET 4.5, WP8, ...)
* .NET 4 Build
* Silverlight Build
* .NET Standard 1.3 Build

I guess that would work, right? The only downside is another `#if` compiler directive to branch for the .NET Standard 1.3 build, which would include the `RegisterProvider` call, right?

Motivation: Right now the .NET Standard 1.0 build is sharing a lot of properties with the PCL Build (actually, it is the same). So for, e.g., .NET 4.5 there are two options. Now we would increase the level of .NET Standard for the better in higher (e.g., .NET 4.6) platforms or compatible .NET Core runtimes. Sounds like a good idea to me...

Any other opinions? Something I am missing here?

**Comment by gabriele-tomassetti** [comment]
Yes, to make it work we just need something like:

```
#if NETSTANDARD1_3
    Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
#endif
```
before any call to `TextEncoding.GetEncoding(String name)`. I made a quick and dirty test in local and it works.

**Comment by iSazonov** [comment]
`project.json` file requires more changes. The complete list of dependencies from [System.Text.Encoding.CodePages](https://www.nuget.org/packages/System.Text.Encoding.CodePages/):
>.NETStandard 1.3 
Microsoft.NETCore.Platforms (>= 1.1.0) 
System.Collections (>= 4.3.0) 
System.Globalization (>= 4.3.0) 
System.IO (>= 4.3.0) 
System.Reflection (>= 4.3.0) 
System.Resources.ResourceManager (>= 4.3.0) 
System.Runtime (>= 4.3.0) 
System.Runtime.Extensions (>= 4.3.0) 
System.Runtime.Handles (>= 4.3.0) 
System.Runtime.InteropServices (>= 4.3.0) 
System.Text.Encoding (>= 4.3.0) 
System.Threading (>= 4.3.0) 

And with move to NetStandard 1.3 you lost Windows 2003/XP support.

**Comment by FlorianRappl** [comment]
> And with move to NetStandard 1.3 you lost Windows 2003/XP support.

That is actually a great remark. Potentially, the best way is to have multiple binaries (as we do now) which include the current .NET Standard 1.0 and then .NET Standard 1.3 (or even 2.0).

I now set this to the v1.0 release and I hope we figure out a good strategy until then.

I do believe that the library should have the largest possible user-base / framework support even though that means additional complexity. Of course, there is a threshold before the gain would be minimal and the effort would be maximal.

**Comment by alexanderuv** [comment]
This was addressed by #724

**Comment by FlorianRappl** [comment]
Thanks @alexanderuv !
